src/segment-loader.js

```diff
@@ -327,12 +331,42 @@ export default class SegmentLoader extends videojs.EventTarget {
     let startOfSegment = duration(playlist,
                                   playlist.mediaSequence + mediaIndex,
                                   this.expired_);
-    let segmentRange = videojs.createTimeRanges([[
-      Math.max(currentTime, startOfSegment),
-      startOfSegment + segment.duration
+    let endOfSegment = startOfSegment + segment.duration;
+
+    // The entire time range of the segment
+    let originalSegmentRange = videojs.createTimeRanges([[
+      startOfSegment,
+      endOfSegment
+    ]]);
+
+    // The adjusted segment time range that is setup such that it starts
+    // no earlier than currentTime
+    // Flash has no notion of a back-buffer so adjustedSegmentRange adjusts
```

Comment:
We shouldn't be apply Flash-specific fixes in this project. Can we move this into the Flash tech or media sources?